src/Microsoft.Windows.ComputeVirtualization/Container.cs

```diff
@@ -25,44 +26,92 @@ public class ProcessStartInfo
     /// </summary>
     public class Container : IDisposable
     {
-        private string _id;
+        private IntPtr _cs;
         private bool _killOnClose;
-        private bool _killed;
+        private bool _dead;
 
-        internal Container(string id, bool terminateOnClose)
+        private HcsNotificationWatcher _watcher;
+
+        private Container(string id, IntPtr computeSystem, bool terminateOnClose, HcsNotificationWatcher watcher)
         {
-            _id = id;
             _killOnClose = terminateOnClose;
+            _cs = computeSystem;
+            _watcher = watcher;
+        }
+
+        internal static Container Initialize(string id, IntPtr computeSystem, bool terminateOnClose)
+        {
+            var watcher = new HcsNotificationWatcher(
+                computeSystem,
+                HcsFunctions.HcsRegisterComputeSystemCallback,
+                HcsFunctions.HcsUnregisterComputeSystemCallback,
+                new HCS_NOTIFICATIONS[]{
+                    HCS_NOTIFICATIONS.HcsNotificationSystemExited,
+                    HCS_NOTIFICATIONS.HcsNotificationSystemCreateCompleted,
+                    HCS_NOTIFICATIONS.HcsNotificationSystemStartCompleted,
+                    HCS_NOTIFICATIONS.HcsNotificationServiceDisconnect
+                    }
+                );
+            var container = new Container(
+                id,
+                computeSystem,
+                terminateOnClose,
+                watcher);
+            watcher.Wait(HCS_NOTIFICATIONS.HcsNotificationSystemCreateCompleted);
+
+            return container;
         }
 
         /// <summary>
         /// Starts the container.
         /// </summary>
         public void Start()
         {
-            HcsFunctions.StartComputeSystem(_id);
+            StartAsync().Wait();
+        }
+
+        public async Task StartAsync()
+        {
+            string result;
+            HcsFunctions.ProcessHcsCall(HcsFunctions.HcsStartComputeSystem(_cs, null, out result), result);
```

Comment:
Its interesting that all the other places you do a 
```C#
if (ProcessHcsCall)
{
    await 
}
```
but here you chose not to. Was that oversight or on purpose?